embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java

```diff
@@ -15,19 +18,17 @@ public class PgVectorEmbeddingStoreRecorder {
 
     public Function<SyntheticCreationalContext<PgVectorEmbeddingStore>, PgVectorEmbeddingStore> embeddingStoreFunction(
             PgVectorEmbeddingStoreConfig config, String datasourceName) {
-        return new Function<>() {
-            @Override
-            public PgVectorEmbeddingStore apply(SyntheticCreationalContext<PgVectorEmbeddingStore> context) {
-                AgroalDataSource dataSource;
-                if (datasourceName == null) {
-                    dataSource = context.getInjectedReference(AgroalDataSource.class, new Default.Literal());
-                } else {
-                    dataSource = context.getInjectedReference(AgroalDataSource.class, new DataSourceLiteral(datasourceName));
-                }
-                return new PgVectorEmbeddingStore(dataSource, config.table(), config.dimension(), config.useIndex(),
-                        config.indexListSize(), config.createTable(), config.dropTableFirst());
-            }
+        return context -> {
+            AgroalDataSource dataSource = Optional.ofNullable(datasourceName)
+                    .map(DataSourceLiteral::new)
+                    .map(dl -> context.getInjectedReference(AgroalDataSource.class, dl))
+                    .orElse(context.getInjectedReference(AgroalDataSource.class, new Default.Literal()));
+
+            dataSource.flush(AgroalDataSource.FlushMode.GRACEFUL);
```

Comment:
The first build [failed](https://github.com/quarkiverse/quarkus-langchain4j/actions/runs/9214412003/job/25350530831), i think it's link to the fact we are using FlushMode.GRACEFUL, perhaps FlushMode.ALL will be better. WDYT ?